google/cloud/functions/internal/parse_cloud_event_json_test.cc

```diff
@@ -207,6 +207,11 @@ TEST(ParseCloudEventJson, WithDataBase64Padding) {
     "source" : "/mycontext",
     "id" : "A234-1234-1234",
     "data_base64" : "YQ=="})js"},
+      {"a", R"js({
+    "type" : "com.example.someevent",
+    "source" : "/mycontext",
+    "id" : "A234-1234-1234",
+    "data_base64" : "YQ="})js"},
```

Comment:
It might be nice to separate this case (like `kExcessivePadding`), or at least comment it as the "under-padded" version of `"a"`.

There is also an argument that this should throw `invalid_argument` too, but your choice.

Reply:
Done, I created two other cases of underpadding too.  I prefer to handle underppading gracefully.  Apparently it is not unheard of to drop the `=` pads in some base64 encoders :shrug: